Sources/RInAppMessaging/TooltipDispatcher.swift

```diff
@@ -113,18 +113,22 @@ internal class TooltipDispatcher: TooltipDispatcherType, ViewListenerObserver {
                     return shouldShow
                 }(),
                 completion: { cancelled in
-                    self.dispatchQueue.async {
-                        if !cancelled {
-                            self.campaignRepository.decrementImpressionsLeftInCampaign(id: tooltip.id)
-                        }
-                        self.queuedTooltips.remove(tooltip)
-                    }
+                    self.handleDisplayCompletion(cancelled: cancelled, tooltip: tooltip)
                 }
             )
             waitForImageDispatchGroup.leave()
         }
     }
 
+    private func handleDisplayCompletion(cancelled: Bool, tooltip: Campaign) {
```

Comment:
nit: This function was required by SonarQube?

Reply:
Yes. To reduce complexity (nested closure levels)